spec/spec.md

```diff
@@ -163,11 +163,28 @@ top-level format, as defined below:
 
 ### Secure Wrapper
 
-::: todo
-Objects need to use a standard means of encryption. Some considerations are:
-- Must account for our multi-recipient access requirements
-- Try to minimize need for re-encryption when permissions change
-:::
+The identity hub data structure uses DagJOSE as the security wrapper for both signatures and encryption. JOSE is very flexible when it comes to the specific algorithm to use. Furthermore it also supports multiple signatories and multiple recipients when dealing with signed and encrypted data respectively.
+
+#### Signature envelope
+Signed objects are encoded using DagJWS, which requires the payload to be encoded as a CID. The payload itself is stored separately using DagCBOR, or some other IPLD codec. The JWS protected header should include a `kid` property with the value being the DIDUrl of the key that signed the JWS.
+
+#### Encryption envelope
+Encrypted objects are encoded using DagJWE. In order to properly encrypt data with DagJWE the data needs to bw encoded as an *inline CID* (see details below). JWEs can be used for both symmetric and asymmetric encryption, at the wrapper layer we don't make any prescriptions about this. JWEs can also use various different ciphers, we reccomend the use of `xchacha20poly1305` as it is one of the most performat ciphers and there are multiple implementations.
+
+##### Encoding an *inline CID*
+In order to encode an object as an *inline CID* the following algorithm is used:
+
+1. Encode the object using some IPLD codec (e.g. DagCBOR)
```

Comment:
```suggestion
1. Encode the object using some IPLD codec (e.g., DagCBOR)
```

Reply:
This seems weird.

Reply:
@oed if it's not a suitable change, you can decline the change, which allows us to merge

Reply:
All good to merge on my end.

Reply:
See https://www.grammarly.com/blog/know-your-latin-i-e-vs-e-g/